src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java

```diff
@@ -0,0 +1,223 @@
+import io.cdap.plugin.gcp.common.SourceOutputFormatProvider;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.HColumnDescriptor;
+import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.hadoop.hbase.mapreduce.TableOutputFormat;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link BatchSink} that writes data to Cloud Bigtable.
+ * This plugin takes a {@link StructuredRecord} in, converts it to {@link Put} mutation, and writes it to the
+ * Cloud Bigtable instance.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name(BigtableSink.NAME)
+@Description("This sink writes data to Google Cloud Bigtable. " +
+  "Cloud Bigtable is Google's NoSQL Big Data database service.")
+public final class BigtableSink extends BatchSink<StructuredRecord, ImmutableBytesWritable, Mutation> {
+  public static final String NAME = "Bigtable";
+  private static final Logger LOG = LoggerFactory.getLogger(BigtableSink.class);
+
+  private static final Set<Schema.Type> SUPPORTED_FIELD_TYPES = ImmutableSet.of(
+    Schema.Type.BOOLEAN,
+    Schema.Type.INT,
+    Schema.Type.LONG,
+    Schema.Type.FLOAT,
+    Schema.Type.DOUBLE,
+    Schema.Type.BYTES,
+    Schema.Type.STRING
+  );
+
+  private final BigtableSinkConfig config;
+  private RecordToHBaseMutationTransformer transformer;
+
+  public BigtableSink(BigtableSinkConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer configurer) {
+    super.configurePipeline(configurer);
+    config.validate();
+    Schema inputSchema = configurer.getStageConfigurer().getInputSchema();
+    if (null == inputSchema) {
+      throw new InvalidStageException("Input schema was not provided");
+    }
+    validateInputSchema(inputSchema);
+    if (config.connectionParamsConfigured()) {
+      Configuration conf = getConfiguration();
+      try (Connection connection = BigtableConfiguration.connect(conf);
+           Admin admin = connection.getAdmin()) {
+        TableName tableName = TableName.valueOf(config.table);
+        if (admin.tableExists(tableName)) {
+          validateExistingTable(connection, tableName);
+        }
+      } catch (IOException e) {
+        throw new InvalidStageException("Failed to connect to Bigtable", e);
+      }
+    }
+
+    configurer.getStageConfigurer().setOutputSchema(inputSchema);
```

Comment:
there's no need to call this, sinks don't need to set an output schema.

Reply:
fixed